feature: add prettier and eslint

SundayTyper opened the pull request:
Resolves https://github.com/BackStoryOrg/website/issues/6

I added to default configurations of prettier and eslint. We can define rules and standards as we go. 

Currently there are clashes between prettier and eslint, and prettier will run on each commit to maintain standards. 

I also removed some demo files that weren't relevant to BackStory, and added some folders to the directory to better locate files down the line. 

Time spent, around an hour (including other hyperlink ticket)

SundayTyper committed:
changed heading discord link

SundayTyper committed:
added link to body list. added underline

SundayTyper committed:
feature: added pretteir and eslint
chore: Cleaning

rendall commented:
Regarding the conflicts between prettier and eslint, [this page](https://prettier.io/docs/en/integrating-with-linters.html) should help. <https://prettier.io/docs/en/integrating-with-linters.html>  I used it in another project and it works great.

SundayTyper committed:
fix: correct some imports

SundayTyper committed:
fix: other import erros

SundayTyper committed:
fix: import errors

rendall commented:
Check out this article on git commit messages when you have a chance:  <https://cbea.ms/git-commit/>  

This advice is an industry standard.  The main point is to use the _imperative mood_ when writing the subject of a git commit message.

So for instance these commit messages:

- [changed heading discord link](https://github.com/BackStoryOrg/website/pull/7/commits/ca65fef9243175711912827228ad1e39bb6837a5)
- [added link to body list. added underline](https://github.com/BackStoryOrg/website/pull/7/commits/dc592f78fe5a2f04520c994e01bc42b1a6224fa9)
- [feature: added pretteir and eslint](https://github.com/BackStoryOrg/website/pull/7/commits/bc7f8abb831abe705fe34dc8c3f9126ea4aeebda)

could instead be:

- _change heading discord link_
- _add link and underline to body list_
- _add prettier and eslint_

Having a consistent way of creating git messages reduces cognitive load.

rendall commented:
If you think it's valuable, we could add "fix:" and "feature:" labels to branch names. I've never really seen the value, but I don't mind doing it if we think it's best.

rendall commented on .eslintrc.json:
```diff
@@ -1,3 +1,3 @@
 {
-  "extends": "next/core-web-vitals"
+  "extends": ["next/core-web-vitals", "prettier"]
```
👍🏻

rendall commented on src/components/common/Button.tsx:
```diff
@@ -0,0 +1,63 @@
+import Link from "next/link";
```
~~@SundayTyper Did you write this Button component?  We should actually use the Tailwind UI components for consistency.~~

Nevermind, I see that you just moved the components from `/src/components`

@Quest037 I think we should probably give Ed your Tailwindui keys.

Quest037 replied:
@rendall @SundayTyper - tailwind keys will be made available in today's daily :)

rendall commented on .husky/pre-commit:
Let's talk among the team and come to a consensus before adding git scripts, or anything that changes expected behavior.  This script will run every commit, and if the team is not expecting it, could lead to confusion.  It might be a good idea, but let's discuss the problem it is attempting to solve.

As well, `husky` may not be necessary.  I'm reluctant to add dependencies in general.

For now, please take this script out.  To discuss any problem and solution, either write an issue, ping #staff or bring it up at daily.

rendall commented on package.json:
```diff
@@ -27,8 +28,14 @@
-    "prettier": "^3.1.1",
+    "eslint-config-prettier": "^9.1.0",
+    "husky": "^9.0.11",
+    "lint-staged": "^15.2.2",
```
Before adding dependencies, let's discuss the problems that dependencies will solve and brainstorm solutions as a team.

Please take these out for now using `npm remove`

rendall commented on package.json:
```diff
@@ -27,8 +28,14 @@
     "sharp": "0.33.1"
+  },
+  "lint-staged": {
+    "**/*": "prettier --write --ignore-unknown"
```
Please remove this, too.

rendall commented on src/app/layout.tsx:
```diff
@@ -1,45 +1,45 @@
-import { Inter, Lexend } from 'next/font/google'
-import clsx from 'clsx'
+import { Inter, Lexend } from "next/font/google";
+import clsx from "clsx";
```
Git history will show that you have authored these changes, when they're only formatting changes. We want formatting changes to be invisible in the history.  Here's an article about it:  <https://github.blog/changelog/2022-03-24-ignore-commits-in-the-blame-view-beta/>  


I will go over how to fix this with you in a 1:1, so don't worry about doing this on your own.  Messing with git history can get very tricky. Wil will reset the history back to before the formatting was run and add those formatted files to their own commit with a "Format" message.  Then we can add that commit hash to a `.git-blame-ignore-revs` file.

rendall reviewed (changes requested):
I'll go over the requested changes in a 1:1.  For now, please take out the husky and lint-staged dependencies.

Quest037 commented:
@SundayTyper - best not to include me in code reviews bud. I'll end up doing more harm than good. When we get to the stage of considering content, design and wording, tag me in. :)

Quest037 reviewed (commented):
Refer to @rendall on these changes. :)

SundayTyper commented:
> If you think it's valuable, we could add "fix:" and "feature:" labels to branch names. I've never really seen the value, but I don't mind doing it if we think it's best.

I've always been told it's standard/good practice to run PRs and commit messages with those prefixes, so it's now a mix of habit and effort

SundayTyper committed:
add pre-commit hook

SundayTyper committed:
attempt to make bad styling change

SundayTyper committed:
run prettier

SundayTyper committed:
make bad styling change

SundayTyper committed:
styling change

SundayTyper committed:
test pre-push linting

SundayTyper committed:
correct styling

SundayTyper committed:
test prettier config

SundayTyper committed:
fix styling

SundayTyper committed:
add npm run fix
remove lint-staged & hu

SundayTyper committed:
resolve conflict

SundayTyper committed:
Merge branch 'master' of into tree_setup

SundayTyper commented:
Added prettier and es-lint to dev dependencies and added 'npm run fix' command to package.json

Removed previous work on husky package and pre-commit hook. Our use of husky doesn't warrant the use of a package and I had implementation questions over pre-commit, giving heightened permissions to the hook file

rendall commented on package.json:
```diff
@@ -6,7 +6,8 @@
     "start": "next start",
-    "lint": "next lint"
+    "lint": "next lint",
+    "fix": "prettier . --write"
```
```suggestion
    "fix": "prettier . --write && eslint . --fix"
```

rendall reviewed (approved):
Just change the command to include `eslint . --fix` and we're good to go.

`"fix": "prettier . --write && eslint . --fix"`

SundayTyper committed:
add eslint to fix command
command ran
